pkg/tsdb/cloudwatch/get_metric_query_batches_test.go

```diff
@@ -0,0 +1,146 @@
+package cloudwatch
+
+import (
+	"testing"
+
+	"github.com/grafana/grafana/pkg/infra/log/logtest"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetMetricQueryBatches(t *testing.T) {
+	logger := &logtest.Fake{}
+	insight1 := models.CloudWatchQuery{
+		MetricQueryType: models.MetricQueryTypeQuery,
+		Id:              "i1",
+	}
+	insight2 := models.CloudWatchQuery{
+		MetricQueryType: models.MetricQueryTypeQuery,
+		Id:              "i2",
+	}
+	insight3 := models.CloudWatchQuery{
+		MetricQueryType: models.MetricQueryTypeQuery,
+		Id:              "i3",
+	}
+	metricStat := models.CloudWatchQuery{
+		MetricQueryType:  models.MetricQueryTypeSearch,
+		MetricEditorMode: models.MetricEditorModeBuilder,
+		Id:               "s1",
+	}
+	math1 := models.CloudWatchQuery{
+		MetricQueryType:  models.MetricQueryTypeSearch,
+		MetricEditorMode: models.MetricEditorModeRaw,
+		Expression:       "PERIOD(i1)",
+		Id:               "m1",
+	}
+	math2 := models.CloudWatchQuery{
+		MetricQueryType:  models.MetricQueryTypeSearch,
+		MetricEditorMode: models.MetricEditorModeRaw,
+		Expression:       "RATE(i1)",
+		Id:               "m2",
+	}
+	math3 := models.CloudWatchQuery{
+		MetricQueryType:  models.MetricQueryTypeSearch,
+		MetricEditorMode: models.MetricEditorModeRaw,
+		Expression:       "m1 * m2",
+		Id:               "m3",
+	}
+	math4 := models.CloudWatchQuery{
+		MetricQueryType:  models.MetricQueryTypeSearch,
+		MetricEditorMode: models.MetricEditorModeRaw,
+		Expression:       "SUM(s1)",
+		Id:               "m4",
+	}
+	math5 := models.CloudWatchQuery{
+		MetricQueryType:  models.MetricQueryTypeSearch,
+		MetricEditorMode: models.MetricEditorModeRaw,
+		Expression:       "PERIOD(i1) * RATE(i3)",
+		Id:               "m5",
+	}
+
+	t.Run("zero insight queries should not batch", func(t *testing.T) {
```

Comment:
`batch` here may not be clear whether it refers to "grouping together" or "separating" -- maybe we could be a little more explicit which one it is in each test title?